### PR TITLE
Respect cargo offline env variable in wasm builder

### PR DESCRIPTION
**Pull request by skunert**
When no internet connection is available, our build scripts fail because they launch processes that attempt to update the `crates.io` index, even when cargo is running with `--offline` or `CARGO_NET_OFFLINE=true`. 

In this PR we look for `CARGO_NET_OFFLINE` and provide `--offline` to the launched build and metadata processes.

Example: `CARGO_NET_OFFLINE=true cargo build`

### CODE REVIEW

File: utils/wasm-builder/src/lib.rs [hunk 1]
```diff
@@ -119,6 +119,9 @@ pub use builder::{WasmBuilder, WasmBuilderSelectProject};
 /// Environment variable that tells us to skip building the wasm binary.
 const SKIP_BUILD_ENV: &str = "SKIP_WASM_BUILD";
 
+/// Environment variable that tells us whether we should avoid network requests
+const OFFLINE: &str = "CARGO_NET_OFFLINE";
```
Comment (bkchr) [inline utils/wasm-builder/src/lib.rs:123]: The crate docs should be updated to mention the env variable.

File: utils/wasm-builder/src/wasm_project.rs [hunk 2]
```diff
@@ -631,6 +631,10 @@ fn build_project(
 	build_cmd.arg("--profile");
 	build_cmd.arg(profile.name());
 
+	if Ok("true".to_string()) == env::var(OFFLINE) {
```
Comment (bkchr) [inline utils/wasm-builder/src/wasm_project.rs:634]:
```suggestion
	if env::var(OFFLINE)map_or(false, |v| v == "true") {
```

File: utils/wasm-builder/src/wasm_project.rs [hunk 3]
```diff
@@ -751,6 +755,16 @@ impl<'a> Deref for DeduplicatePackage<'a> {
 	}
 }
 
+fn create_metadata_command(path: impl Into<PathBuf>) -> MetadataCommand {
+	let mut metadata_command = MetadataCommand::new();
+	metadata_command.manifest_path(path);
+
+	if Ok("true".to_string()) == env::var(OFFLINE) {
```
Comment (bkchr) [inline utils/wasm-builder/src/wasm_project.rs:762]:
```suggestion
	if env::var(OFFLINE)map_or(false, |v| v == "true") {
```

**Review by bkchr — APPROVED** [review]
Please also check if `cargo` may not also supports to set the variable to `1` as `true`. If it supports that, we should do the same.

**Comment by davxy** [comment]
Nit. Maybe the "value" associated to the env variable can just be ignored (1, TRUE, true, ...).

If I see correctly for the other boolean variables we just check their presence.

In particular, for `SKIP_BUILD_ENV`
https://github.com/paritytech/substrate/blob/90d08f77aaaebe8f7012216948e54b6684a175b3/utils/wasm-builder/src/builder.rs#L188-L191

And for `WASM_BUILD_NO_COLOR`
https://github.com/paritytech/substrate/blob/90d08f77aaaebe8f7012216948e54b6684a175b3/utils/wasm-builder/src/lib.rs#L289-L291

Maybe we can define 

```
fn offline_build() -> bool {
    env::var(crate::OFFLINE).is_ok()
}
```

**Comment by davxy** [comment]
There's also one more doc to add here 

https://github.com/paritytech/substrate/blob/90d08f77aaaebe8f7012216948e54b6684a175b3/utils/wasm-builder/src/lib.rs#L66

**Review by davxy — APPROVED** [review]

**Comment by skunert** [comment]
bot merge

**Pull request merged**